python/module.hpp

```diff
@@ -68,13 +68,32 @@ void func(const T & self, py::array_t<Real> a, py::array_t<Real> b,
   } else {                                                                        \
     self.func(a.data(), b.data(), out.mutable_data(), num, NoLambdaHelper());      \
   }                                                                               \
-}
-
-#define EOS_VEC_FUNC_NO_LAMBDA_TMPL(func, a, b, out)                              \
+}                                                                                 \
+                                                                                  \
 template<typename T>                                                              \
-void func####NoLambda(const T & self, py::array_t<Real> a, py::array_t<Real> b,               \
+void func####WithScratch(const T & self, py::array_t<Real> a, py::array_t<Real> b,\
+          py::array_t<Real> out, py::array_t<Real> scratch, const int num, py::array_t<Real> lambdas){       \
+  py::buffer_info lambdas_info = lambdas.request();                               \
+  if (lambdas_info.ndim != 2)                                                     \
+      throw std::runtime_error("lambdas dimension must be 2!");                   \
+                                                                                  \
+  if(lambdas_info.shape[1] > 0) {                                                 \
+    self.func(a.data(), b.data(), out.mutable_data(), scratch.mutable_data(), num, LambdaHelper(lambdas)); \
+  } else {                                                                        \
+    self.func(a.data(), b.data(), out.mutable_data(), scratch.mutable_data(), num, NoLambdaHelper());      \
+  }                                                                               \
+}                                                                                 \
```

Comment:
Is the "WithScratch" variant of the functions tested somewhere?

Reply:
@jhp-lanl just added Python tests that do so: https://github.com/lanl/singularity-eos/blob/59a2e74f784d6b1ce36304331893a74f62512c89/test/python_bindings.py#L528